### PR TITLE
Ana/use optional chaining for refs.action modal.show

### PR DESCRIPTION
**Pull request by Bitcoinera**
Closes #ISSUE

**Description:**

A new Era has begun...

Fixes this in ValidatorPage:

image

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/ActionModal/components/UndelegationModal.vue [hunk 1]
```diff
@@ -411,7 +411,11 @@ export default {
       `,
       /* istanbul ignore next */
       skip() {
-        return !this.address ||  !this.$refs.actionModal || !this.$refs.actionModal.show
+        return (
+          !this.address ||
+          !this.$refs.actionModal ||
+          !this.$refs.actionModal?.show
```
Comment (faboweb) [inline src/ActionModal/components/UndelegationModal.vue:417]:
```suggestion
          !this.$refs.actionModal?.show
```

the chaining does exactly reduce the need for that line

Comment (Bitcoinera) [reply src/ActionModal/components/UndelegationModal.vue:417]: true xDDD

**Comment by Bitcoinera** [comment]
I'd need to recheck, but `yarn install` seems now terribly slow? 🤷

File: babel.config.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
 module.exports = {
   presets: [
     [
-      "@vue/app",
+      "@vue/cli-plugin-babel/preset",
```
Comment (jbibla) [inline babel.config.js:4]: what does this do?

**Comment by jbibla** [comment]
@Bitcoinera how does chaining solve this problem?

**Comment by faboweb** [comment]
> @Bitcoinera how does chaining solve this problem?

this doesn't change a problem. this is "refactoring" using an operator to reduce code. The operator is main ECMA script but not yet native in all browsers (already Chrome and others).

**Comment by jbibla** [comment]
> Fixes this in ValidatorPage
?

**Comment by faboweb** [comment]
> yep i see that.

> > Fixes this in ValidatorPage

https://github.com/luniehq/lunie/pull/4184/files#diff-35581415b274540f062489ab1c34e080R491

here is the fix. But it is already fixed on develop

**Comment by Bitcoinera** [comment]
> 
> https://github.com/luniehq/lunie/pull/4184/files#diff-35581415b274540f062489ab1c34e080R491
> 
> here is the fix. But it is already fixed on develop

Should we leave this super mega advancement in Javascript for another time or...

go for it!!!! I can refactor something else. We have been having so many errors due to the same thing

**Comment by faboweb** [comment]
I am very open to doing this! Do we need to switch the babel config? https://github.com/luniehq/lunie/pull/4184/files#diff-fc10a6682269c1fefc2abe05d1500b8aR4

File: babel.config.js (hunk 2 above)
Comment (Bitcoinera) [reply babel.config.js:4]: This is an array of babel plugins that are specifically for Vue. [Here ](https://babeljs.io/docs/en/presets)it explains what presets are (array of plugins so you don't need to configure them yourself)

We need this in order to have all the required plugins in our system to use ECMAScript 2020

**Comment by Bitcoinera** [comment]
> I am very open to doing this! Do we need to switch the babel config? https://github.com/luniehq/lunie/pull/4184/files#diff-fc10a6682269c1fefc2abe05d1500b8aR4

It seems so. I found this specific way to do it for Vue [here](https://code.luasoftware.com/tutorials/vuejs/vue-cli-project-enable-optional-chaining/).  

Babel is what is compiling our Javascript so it makes sense we need to change the config so we are able to make our JS "smarter"

**Comment by Bitcoinera** [comment]
Weird that the unit tests worked in the beginning and after merge they don't. Same than why suddenly  the levenary module was missing.

Probably dependabot changed something?

**Comment by Bitcoinera** [comment]
Still another babel module? 🤦

**Comment by Bitcoinera** [comment]
I think something is wrong... so weird. Why it was OK in the beginning and now there so many missing modules. What changed? 🤔
It is like it is not getting all the plugins it needs from @vue/cli-plugin-babel/preset

**Comment by faboweb** [comment]
Let's skip this for now, as it seems to be more complicated than a simple change

**Comment by Bitcoinera** [comment]
> Let's skip this for now, as it seems to be more complicated than a simple change

😓 so weird that it suddenly changed.

Was super excited to have optional chaining!

But hey, it will be much easier in the BE. We can add it there for now 🤩

**Pull request closed without merging**